sdk/servicebus/service-bus/src/serializers/ruleResourceSerializer.ts

```diff
@@ -0,0 +1,326 @@
+   */
+  sqlExpressionFilter?: string;
+
+  /**
+   * Defines the expression that the rule evaluates. Only the messages whose CorrelationId match the CorrelationId set in the filter expression are allowed. Only one between a correlation and a sql expression can be defined.
+   */
+  correlationFilter?: CorrelationFilterOptions;
+
+  /**
+   * Defines the expression that the rule evaluates as a true filter.
+   */
+  trueFilter?: string;
+
+  /**
+   * Defines the expression that the rule evaluates as a false filter.
+   */
+  falseFilter?: string;
+
+  /**
+   * Defines the expression that the rule evaluates. If the rule is of type SQL, the expression string is interpreted as a SQL92 expression which must evaluate to True or False. If the rule is of type CorrelationFilterExpression then only the messages whose CorrelationId match the CorrelationId set in the filter expression are allowed.
+   */
+  sqlRuleAction?: string;
+}
+
+/**
+ * Represents all attributes of a rule entity
+ */
+export interface Rule {
+  /**
+   * Name of the rule
+   */
+  ruleName?: string;
+
+  /**
+   * Name of topic
+   */
+  topicName?: string;
+
+  /**
+   * Name of subscription
+   */
+  subscriptionName?: string;
+
+  /**
+   * Represents Topic filter.
+   */
+  filter?: Filter;
+
+  /**
+   * Rule action
+   */
+  action?: string;
+
+  /**
+   * Created at timestamp
+   */
+  createdAt?: string;
+}
+
+/**
+ * Represents all possible fields on Topic Filter
+ */
+export interface Filter extends CorrelationFilterOptions {
+  /**
+   * SQL expression to use.
+   */
+  sqlExpression?: string;
+
+  /**
+   * CompatibilityLevel field on the SQL rule
+   */
+  compatibilityLevel: number;
+}
+
+/**
+ * Represents settable options on CorrelationFilter
+ */
+export interface CorrelationFilterOptions {
+  /**
+   * CorrelationId value
+   */
+  correlationId?: string;
+
+  /**
+   * Message label
+   */
+  label?: string;
+
+  /**
+   * Message ID
+   */
+  messageId?: number;
+
+  /**
+   * Message 'To' field
+   */
+  to?: number;
+
+  /**
+   * Message 'ReplyTo' field
+   */
+  replyTo?: number;
+
+  /**
+   * Message 'SessionId' field
+   */
+  sessionId?: string;
+
+  /**
+   * Message 'ReplyToSessionId' field
+   */
+  replyToSessionId?: string;
+
+  /**
+   * Message 'ContentType' field
+   */
+  contentType?: string;
+
+  /**
+   * Custom key value properties.
+   */
+  properties?: any;
+}
+
+/**
+ * @ignore RuleResourceSerializer for serializing / deserializing Rule entities
+ */
+export class RuleResourceSerializer implements AtomXmlSerializer {
+  serialize(rule: InternalRuleOptions): string {
+    const resource: { Name: any; Filter: any; Action: any } = {
+      Name: rule.name,
+      Filter: [],
+      Action: []
+    };
+
+    if (rule) {
+      const filters = [];
```

Comment:
Instead of creating a separate `filters` array, you could push your filters directly to `resource.Filter`. Same applies to `actions`.

Reply:
Not expected to be an array